sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -81,7 +81,9 @@ public void close() {
   }
 
   public void reserve(int requiredCapacity) {
-    if (requiredCapacity > capacity) {
+    if (requiredCapacity < 0) {
+      throwUnsupportedException(requiredCapacity, null);
```

Comment:
how about use `ArithmeticException` instead of null?

Reply:
@adrian-wang Since this is just a backport, I would prefer to keep the modifications of the original commit to the minimum. Also, it might be a bit confusing the user, if we create `ArithmeticException` here, as this is not the place where invalid computation was performed, but just the place where we realize the input is not sane.

Reply:
thanks for the explanation!